docs/src/main/sphinx/connector/sqlserver.rst

```diff
@@ -366,6 +366,46 @@ For example, select the top 10 percent of nations by population::
         )
       );
 
+.. _sqlserver-procedure-function:
+
+``procedure(varchar) -> table``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The ``procedure`` function allows you to run stored procedures on the underlying
+database directly.
+
+.. note::
+
+    The ``procedure`` function does not support running StoredProcedures that return multiple statements,
+    use a non-select statement, use output parameters, or use conditional statements.
+
+
+The follow example runs the stored procedure ``employee_sp`` in the ``example`` catalog and the
+``example_schema`` in the underlying SQL Server database::
+
+    SELECT
+      *
+    FROM
+      TABLE(
+        example.system.procedure(
+          schema    => 'example_schema',
+          procedure => 'employee_sp'
+        )
+      );
+
+If the stored procedure ``employee_sp`` requires any input, it can be specified via ``inputs``
+argument as an ``array<varchar>``::
```

Comment:
what if procedure expects an integer?

Reply:
For now we pass the arguments as string - we pass it without quotes for integer and with additional quotes for string (if requested by the underlying datasource).

Reply:
Should we document this?

Reply:
But for SQLServer, it doesn't expect additional quotes. So do we need to document it ?

Reply:
It’d be better to pass the arguments as a ROW, so each one has the proper type. You get type safety and avoid conversions that might not be possible to do easily or reliably.

Reply:
Also, I’m concerned about the “for now” in your statement above. This is the kind of thing that would break backward compatibility if we change it later.

Reply:
Regarding `ROW` type - is it possible to have them dynamically resolved ? Like we don't know how many arguments we would pass to a given stored procedure

Reply:
@martint , I agree with your comment and I would like to address it. However I spent some time on it and I see it is challenging as we need to resolve the fields based on the stored procedure name. Considering that this feature is experimental and is going to address a niche use case I would like to continue with this PR and merge. Later on, once we learn how this feature is used and what is needed I would like us to iterate on it.

Please let me know what you think on the above. I am going to merge it soon.

cc: @kokosing

Reply:
No I was asking about how should we pass the input arguments to the table function - If we need to consider it as a `ScalarArgument` then we need to know the full type definition (i.e which depends on the function to be invoked) or is there a different `ArgumentType` to handle these situations ?

Reply:
The type of scalar argument is established at TF declaration.

Reply:
Even if we could pass arbitrary scalars, it seems right to use an array of varchar in this case. It shifts the responsibility of formatting them the right way to the user. That's in the spirit of query pass-through. This way it is harder for the user to pass parameters, or non-literal expressions, but I don't know if a real use-case would need those capabilities.

Like @kokosing noticed, we must watch out for possible abuse (injection).

Reply:
Table argument is the only argument type of table functions that is _polymorphic_ . Technically, we could make use of it, and pass a scalar subquery containing parameters. But then we'd have to refactor the PTF to use the operator-based execution.